src/dbus/gnome_shell_screenshot.rs

```diff
@@ -39,6 +43,9 @@ impl Screenshot {
             Ok(NiriToScreenshot::ScreenshotResult(None)) => {
                 return Err(fdo::Error::Failed("internal error".to_owned()));
             }
+            Ok(NiriToScreenshot::ColorResult(_)) => {
+                return Err(fdo::Error::Failed("unexpected color result".to_owned()));
+            }
```

Comment:
So this is actually problematic because it can happen if screenshot and pick color are started at once, then the wrong task is woken up. Pick color should use a different channel instead, perhaps it could simply use the same channel used by regular niri IPC?